### PR TITLE
Various Fixes

### PR DESCRIPTION
**Pull request by carturn**
Fixing some issues that have appeared when testing the Lidar->Teensy->RoboRIO system.

### CODE REVIEW

File: TeensyLidarTest/TeensyLidarTest.ino [hunk 1]
```diff
@@ -42,7 +42,7 @@ void loop() {
         error = (current_packet[data_start + 1] & 0x80) > 0;
         if (!error) {
           uint16_t distance = 0;
-          distance = ((current_packet[data_start]) | (current_packet[data_start + 1] & 0x0F) << 8);
+          distance = ((current_packet[data_start]) | (current_packet[data_start+1] & 0x0F) << 8);
```
Comment (leijurv) [inline TeensyLidarTest/TeensyLidarTest.ino]: Why?

Comment (carturn) [reply TeensyLidarTest/TeensyLidarTest.ino]: Arduino autoformat.

File: TeensyLidar/TeensyLidar.ino [hunk 2]
```diff
@@ -74,6 +75,8 @@ void loop() {
     }
   }
 
+  writeLongs(0x607, 0, lidarSpeed);
```
Comment (leijurv) [inline TeensyLidar/TeensyLidar.ino]: magic number

File: TeensyLidar/TeensyLidar.ino [hunk 3]
```diff
@@ -61,10 +61,11 @@ void writeLongs(uint32_t id, long value1, long value2) {
...
 
 void loop() {
-  CAN_update();
-  try_load_next_byte();
+  long loopStart = micros();
 
-  if (!start) {
+  try_load_next_bytes();
+
+  if (subpacket_idx == 21) {
```
Comment (leijurv) [inline TeensyLidar/TeensyLidar.ino]: magic number

**Comment by carturn** [comment]
This has been tested and seems to work at the capacity of the sensor.

**Pull request merged**